src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Windows.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Net;
+using System.Collections;
+using System.ComponentModel;
+using System.Text;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Xml;
+using System.Threading;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.DirectoryServices.Protocols
+{
+    public partial class LdapConnection
+    {
+        private void InternalInitConnectionHandle(string hostname)
+        {
+            // User wants to setup a connectionless session with server.
+            if (((LdapDirectoryIdentifier)_directoryIdentifier).Connectionless == true)
```

Comment:
There are three different hard casts to LdapDirectoryIdentifier... save the hard cast as a local once.